src/neptune/cli/sync.py

```diff
@@ -50,60 +51,122 @@
+                    )
+                    for offline_container in containers.offline_containers
+                ]
+                for future in concurrent.futures.as_completed(futures):
+                    future.result()
 
     @staticmethod
     def sync_selected(
-        *, backend: "NeptuneBackend", base_path: Path, project_name: Optional[str] = None, object_names: Sequence[str]
+        *,
+        backend: "NeptuneBackend",
+        base_path: Path,
+        project_name: Optional[str] = None,
+        object_names: Sequence[str],
+        num_threads: Optional[int] = None,
     ) -> None:
         containers = collect_containers(path=base_path, backend=backend)
-        async_selected = [QualifiedName(name) for name in object_names if not name.startswith(OFFLINE_NAME_PREFIX)]
-
-        if async_selected:
-            sync_selected_async(
-                backend=backend,
-                base_path=base_path,
-                container_names=async_selected,
-                containers=containers.async_containers,
-            )
-
-        offline_selected = [
+        if async_selected := [QualifiedName(name) for name in object_names if not name.startswith(OFFLINE_NAME_PREFIX)]:
+            with concurrent.futures.ThreadPoolExecutor(max_workers=num_threads) as executor:
+                futures = [
+                    executor.submit(
+                        sync_selected_async,
+                        backend=backend,
+                        base_path=base_path,
+                        container_names=[selected],
+                        containers=containers.async_containers,
+                    )
+                    for selected in async_selected
+                ]
+                for future in concurrent.futures.as_completed(futures):
+                    future.result()
+
+        if offline_selected := [
             UniqueId(name[len(OFFLINE_NAME_PREFIX) :]) for name in object_names if name.startswith(OFFLINE_NAME_PREFIX)
-        ]
-        if offline_selected:
-            sync_selected_offline(
-                backend=backend,
-                base_path=base_path,
-                container_names=offline_selected,
-                containers=containers.offline_containers,
-                project_name=project_name,
-            )
+        ]:
+            with concurrent.futures.ThreadPoolExecutor(max_workers=num_threads) as executor:
+                futures = [
+                    executor.submit(
+                        sync_selected_offline,
+                        backend=backend,
+                        base_path=base_path,
+                        container_names=[selected_offline],
+                        containers=containers.offline_containers,
+                        project_name=project_name,
+                    )
+                    for selected_offline in offline_selected
+                ]
+            for future in concurrent.futures.as_completed(futures):
+                future.result()
```

Comment:
I feel that this exact piece of logic (with slightly different parameters each time) is repeated so often it deserves a separate function ;)

Reply:
Will create a reusable function if we stick with this approach :)